src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSMessagingUtils.cs

```diff
@@ -89,33 +89,23 @@ internal static PropagationContext ExtractParentContext(SNSEvent.SNSMessage? mes
 
     private static IEnumerable<string>? SqsMessageAttributeGetter(IDictionary<string, SQSEvent.MessageAttribute> attributes, string attributeName)
     {
-        if (!attributes.TryGetValue(attributeName, out var attribute))
-        {
-            return null;
-        }
-
-        return attribute?.StringValue != null ?
-            new[] { attribute.StringValue } :
+        return !attributes.TryGetValue(attributeName, out var attribute) ? null :
+            attribute?.StringValue != null ? new[] { attribute.StringValue } :
             attribute?.StringListValues;
     }
 
     private static IEnumerable<string>? SnsMessageAttributeGetter(IDictionary<string, SNSEvent.MessageAttribute> attributes, string attributeName)
     {
-        if (!attributes.TryGetValue(attributeName, out var attribute))
-        {
-            return null;
-        }
-
-        switch (attribute?.Type)
-        {
-            case SnsAttributeTypeString when attribute.Value != null:
-                return new[] { attribute.Value };
-            case SnsAttributeTypeStringArray when attribute.Value != null:
-                // Multiple values are stored as CSV (https://docs.aws.amazon.com/sns/latest/dg/sns-message-attributes.html).
-                return attribute.Value.Split(',');
-            default:
-                return null;
-        }
+        return !attributes.TryGetValue(attributeName, out var attribute)
+            ? null
+            : attribute?.Type switch
+            {
+                SnsAttributeTypeString when attribute.Value != null => (IEnumerable<string>?)[attribute.Value],
```

Comment:
Is the IEnumerable cast really required here? The old `new[]` looked more readable (and shorter) to me in this instance...

Reply:
Fixed in dcda60f6381b9cfea4ff88a5966dc377573f1987